keras_nlp/models/bert/bert_models.py

```diff
@@ -259,215 +215,83 @@ def get_config(self):
                 "max_sequence_length": self.max_sequence_length,
                 "num_segments": self.num_segments,
                 "dropout": self.dropout,
-                "cls_token_index": self.cls_token_index,
             }
         )
         return config
 
+    # TODO(jbischof): consider exposing `presets` as member variable.
+    # Need to protect dict from mutation.
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        load_weights=True,
+        name=None,
```

Comment:
I think `name` and `trainable` makes sense, as opposed to `kwargs`. But it does feel slightly weird that we take `trainable` in explicitly here and implicitly in every other constructor in the library.

The other two main layer params I can think of are `dtype` (which doesn't actually do anything on models today), and `dynamic` (which can be set via `run_eagerly` property for models I think), so I think we are all good there.

cc @fchollet. Are there any other base class init args we should consider here?

Reply:
I actually did this at Francois' suggestion originally for the config-in-code classes. And if you look above, these args are explicit in the class constructor as well!

Reply:
Totally! I was agreeing that you probably have the right approach here above.

I was more asking...

1) if we should consider any extra base layer arguments.
2) if we should do anything about the fact that every other layer in KerasNLP does not take `trainable` explicitly (same with core Keras for that matter). Probably the easiest thing is to say we will only do this explicit `kwargs` enumeration for applications, and not our modular layer offering.

Reply:
Given that you can just do `layer.trainable` after instantiation I'd favor not including it here and ask the user to set the property on the model. We should keep the `from_preset` minimal.

Reply:
For further context on this -- it's generally fine to not expose `trainable` as an explicit constructor arg anywhere (though passing `**kwargs` through in a constructor in always a good idea). For instance Keras Applications never had a `trainable` constructor arg.

Reply:
OK @fchollet it sounds like I should move back to the old `**kwargs` pattern and remove the `name` and `trainable` args. I think before you were reacting to the config-in-code classes having `kwargs` as the only param.